### ISSUE TITLE
Papa not working in electron

### ISSUE THREAD

**Issue by svbeon** [mholt/PapaParse#391]
Hello,

Papa isn't recognized in electron with the plain error
```
Uncaught ReferenceError: Papa is not defined
    at <anonymous>:1:1
```

While I'm not able to debug further (due to time and the fact that I don't know the codebase)
 but here's an example electron app to demonstrate [PapaTest.zip](https://github.com/mholt/PapaParse/files/1055539/PapaTest.zip)
(`unzip PapaTest.zip && cd PapaTest && npm i && electron .`)

**Comment by svbeon** [comment]
on first glance it seems electrons partial support for modules is causing the problem, commenting out the following solves it. But other modules i use that support commonJS doesn't do this, why just Papa?
```js
[...]
else if (typeof module === 'object' && module.exports)
	{
		// Node. Does not work with strict CommonJS, but
		// only CommonJS-like environments that support module.exports,
		// like Node.
		module.exports = factory();
	}
[...]
```

**Comment by pokoli** [comment]
I don't know so much about electron, but as Paparse can be installed as node module I think this is the way to go. 

Have you tried installing them as node object and using it with require?

**Comment by svbeon** [comment]
I've been trying to avoid that as much as possible as I want my code to be
browser-ready, but I could try (i doubt it wouldn't work)

Den tis 6 juni 2017 17:56Sergi Almacellas Abellana <notifications@github.com>
skrev:

**Comment by pokoli** [comment]
The problem is that we support running Papa as node module, and this may cause to break in electron when not using as module (indeed, that's what the code you comented does), so I imagine it will work as node module, as it does on the nodejs console. 

Just try and Let us know how it goes...

**Comment by svbeon** [comment]
Yes, well that's kinda the case, if I have papaparse as a node_module of
the electron app it will work to require('papaparse'), but if I store it
elsewhere (like in `/app`) it won't. But the point is that it's a bit odd
that papa's way to check if modules are supported gives this result, while
others doesn't.

Looking at the code I think it depends on the if statement,

While other libraries (here store.js is our reference) do the check with
```js
if ("object" == typeof exports && "undefined" != typeof module)
```
Papa doesn't check the typeof exports, changing Papa's if check to do
`typeof exports !== 'undefined'` instead of just `module.exports` solves
the problem

Den tis 6 juni 2017 18:02Sergi Almacellas Abellana <notifications@github.com>
skrev:

**Comment by pokoli** [comment]
I just released 4.3.3 which contains this fix, so it's easier to test when installing using npm. Hope it helps!

**Comment by suavelizard** [comment]
I have a similar problem `Uncaught ReferenceError: module is not defined` when setting worker to true
